From 56a0a83ddb1b7472dbc9762e2aca396605c62085 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 05:37:42 -0800
Subject: [PATCH] purpur File Patches


diff --git a/net/minecraft/world/entity/npc/CatSpawner.java b/net/minecraft/world/entity/npc/CatSpawner.java
index 5401976c4fa86e2db6622743c8e8ea7381f72878..278a4000d06690a41e58a059d54e0044944a51fe 100644
--- a/net/minecraft/world/entity/npc/CatSpawner.java
+++ b/net/minecraft/world/entity/npc/CatSpawner.java
@@ -23,7 +23,7 @@ public class CatSpawner implements CustomSpawner {
     public void tick(ServerLevel level, boolean spawnEnemies) {
         this.nextTick--;
         if (this.nextTick <= 0) {
-            this.nextTick = 1200;
+            this.nextTick = level.purpurConfig.catSpawnDelay; // Purpur - Cat spawning options
             Player randomPlayer = level.getRandomPlayer();
             if (randomPlayer != null) {
                 RandomSource randomSource = level.random;
@@ -45,9 +45,12 @@ public class CatSpawner implements CustomSpawner {
     }
 
     private void spawnInVillage(ServerLevel level, BlockPos pos) {
-        int i = 48;
-        if (level.getPoiManager().getCountInRange(holder -> holder.is(PoiTypes.HOME), pos, 48, PoiManager.Occupancy.IS_OCCUPIED) > 4L) {
-            List<Cat> entitiesOfClass = level.getEntitiesOfClass(Cat.class, new AABB(pos).inflate(48.0, 8.0, 48.0));
+        // Purpur start - Cat spawning options
+        int range = level.purpurConfig.catSpawnVillageScanRange;
+        if (range <= 0) return;
+        if (level.getPoiManager().getCountInRange(holder -> holder.is(PoiTypes.HOME), pos, range, PoiManager.Occupancy.IS_OCCUPIED) > 4L) {
+            List<Cat> entitiesOfClass = level.getEntitiesOfClass(Cat.class, new AABB(pos).inflate(range, 8.0, range));
+        // Purpur end - Cat spawning options
             if (entitiesOfClass.size() < 5) {
                 this.spawnCat(pos, level, false);
             }
@@ -55,8 +58,11 @@ public class CatSpawner implements CustomSpawner {
     }
 
     private void spawnInHut(ServerLevel level, BlockPos pos) {
-        int i = 16;
-        List<Cat> entitiesOfClass = level.getEntitiesOfClass(Cat.class, new AABB(pos).inflate(16.0, 8.0, 16.0));
+        // Purpur start - Cat spawning options
+        int range = level.purpurConfig.catSpawnSwampHutScanRange;
+        if (range <= 0) return;
+        List<Cat> entitiesOfClass = level.getEntitiesOfClass(Cat.class, new AABB(pos).inflate(range, 8.0, range));
+        // Purpur end - Cat spawning options
         if (entitiesOfClass.isEmpty()) {
             this.spawnCat(pos, level, true);
         }
