From 56a0a83ddb1b7472dbc9762e2aca396605c62085 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 05:37:42 -0800
Subject: [PATCH] purpur File Patches


diff --git a/net/minecraft/world/entity/projectile/hurtingprojectile/WitherSkull.java b/net/minecraft/world/entity/projectile/hurtingprojectile/WitherSkull.java
index 31125bf31c2e1e749873a0387aea6040f5f832dd..1020e835f24bc25e032335f739d526bc0bd4ecc4 100644
--- a/net/minecraft/world/entity/projectile/hurtingprojectile/WitherSkull.java
+++ b/net/minecraft/world/entity/projectile/hurtingprojectile/WitherSkull.java
@@ -94,7 +94,7 @@ public class WitherSkull extends AbstractHurtingProjectile {
         super.onHit(result);
         if (!this.level().isClientSide()) {
             // CraftBukkit start
-            org.bukkit.event.entity.ExplosionPrimeEvent event = new org.bukkit.event.entity.ExplosionPrimeEvent(this.getBukkitEntity(), 1.0F, false);
+            org.bukkit.event.entity.ExplosionPrimeEvent event = new org.bukkit.event.entity.ExplosionPrimeEvent(this.getBukkitEntity(), this.level().purpurConfig.witherExplosionRadius, false); // Purpur - Config for wither explosion radius
             if (event.callEvent()) {
                 this.level().explode(this, this.getX(), this.getY(), this.getZ(), event.getRadius(), event.getFire(), Level.ExplosionInteraction.MOB);
             }
@@ -103,6 +103,13 @@ public class WitherSkull extends AbstractHurtingProjectile {
         }
     }
 
+    // Purpur start - Add canSaveToDisk to Entity
+    @Override
+    public boolean canSaveToDisk() {
+        return false;
+    }
+    // Purpur end - Add canSaveToDisk to Entity
+
     @Override
     protected void defineSynchedData(SynchedEntityData.Builder builder) {
         builder.define(DATA_DANGEROUS, false);
