From 641e5706903b49be5e7541af8d07dd1dea4ba4e4 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 05:37:42 -0800
Subject: [PATCH] purpur paperServer File Patches


diff --git a/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java b/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java
index c8583cb0bc5a1ce00fa96173097434ed9d1cf781..9fa130bf030bb829a087c2ed9ece99c90182f74e 100644
--- a/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java
+++ b/src/main/java/com/destroystokyo/paper/entity/ai/MobGoalHelper.java
@@ -148,6 +148,11 @@ public class MobGoalHelper {
     private static final Map<String, String> RENAMES = Util.make(new HashMap<>(), map -> {
         map.put("AbstractSkeleton$1", "AbstractSkeletonMelee");
 
+        // Purpur start - Add option to disable zombie aggressiveness towards villagers
+        map.put("Zombie$1", "ZombieAttackVillager");
+        map.put("Drowned$1", "DrownedAttackVillager");
+        // Purpur end - Add option to disable zombie aggressiveness towards villagers
+
         // remove duplicate
         map.put("TraderLlama$TraderLlamaDefendWanderingTraderGoal", "TraderLlamaDefendWanderingTraderGoal");
         map.put("AbstractIllager$RaiderOpenDoorGoal", "RaiderOpenDoorGoal");
