--- a/net/minecraft/world/entity/monster/Shulker.java
+++ b/net/minecraft/world/entity/monster/Shulker.java
@@ -81,7 +_,7 @@
         Vec3i forwardNormal = Direction.SOUTH.getUnitVec3i();
         return new Vector3f(forwardNormal.getX(), forwardNormal.getY(), forwardNormal.getZ());
     });
-    public static final float MAX_SCALE = 3.0F;
+    private static final float MAX_SCALE = 3.0F;
     private float currentPeekAmountO;
     private float currentPeekAmount;
     private @Nullable BlockPos clientOldAttachPosition;
