From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Djatlasv <djatlasv@users.noreply.github.com>
Date: Sat, 14 Mar 2026 23:24:36 -0600
Subject: [PATCH] Osmium: move GrimAC init after Paper GlobalConfiguration init


diff --git a/net/minecraft/server/dedicated/DedicatedServer.java b/net/minecraft/server/dedicated/DedicatedServer.java
index eb47e675fe7fcea022e049f1d57c7f4d10f300c5..09b4cf0b43d8b553379f451e45fb8b8c6e6b3eb6 100644
--- a/net/minecraft/server/dedicated/DedicatedServer.java
+++ b/net/minecraft/server/dedicated/DedicatedServer.java
@@ -269,12 +269,12 @@ public class DedicatedServer extends MinecraftServer implements ServerInterface
         org.osmium.OsmiumConfig.init((java.io.File) options.valueOf("osmium-settings")); // Osmium
         org.osmium.anticheat.OsmiumAltTracker.init(new java.io.File(".")); // Osmium - alt ban IP tracking
         org.osmium.anticheat.OsmiumChatFilter.init(new java.io.File(".")); // Osmium - chat filter
-        if (org.osmium.OsmiumConfig.grimEnabled) { org.osmium.anticheat.grim.OsmiumGrimLoader.init(); } // Osmium - GrimAC
         // Purpur end - Migrate Setting to reintroduce end void rings
         // Paper start - initialize global and world-defaults configuration
         this.paperConfigurations.initializeGlobalConfiguration(this.registryAccess());
         this.paperConfigurations.initializeWorldDefaultsConfiguration(this.registryAccess());
         // Paper end - initialize global and world-defaults configuration
+        if (org.osmium.OsmiumConfig.grimEnabled) { org.osmium.anticheat.grim.OsmiumGrimLoader.init(); } // Osmium - GrimAC (after Paper config init)
         this.server.spark.enableEarlyIfRequested(); // Paper - spark
         // Paper start - fix converting txt to json file; convert old users earlier after PlayerList creation but before file load/save
         if (this.convertOldUsers()) {
