From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Djatlasv <djatlasv@users.noreply.github.com>
Date: Fri, 10 Apr 2026 09:13:37 -0600
Subject: [PATCH] Osmium: add --osmium-settings CLI option and config reload
 hook


diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index ecc3d78eeaa776e3aef4cfad5ab5c7191f5eed06..142cafa23399b6bb6673d234e94da6add2365c60 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -1005,6 +1005,7 @@ public final class CraftServer implements Server {
         org.spigotmc.SpigotConfig.init((File) this.console.options.valueOf("spigot-settings")); // Spigot
         this.console.paperConfigurations.reloadConfigs(this.console);
         org.purpurmc.purpur.PurpurConfig.init((File) console.options.valueOf("purpur-settings")); // Purpur - Purpur config files
+        org.osmium.OsmiumConfig.init((java.io.File) console.options.valueOf("osmium-settings")); // Osmium
         for (ServerLevel world : this.console.getAllLevels()) {
             // world.serverLevelData.setDifficulty(config.difficulty); // Paper - per level difficulty
             world.setSpawnSettings(world.isSpawningMonsters()); // Paper - per level difficulty (from MinecraftServer#setDifficulty(ServerLevel, Difficulty, boolean))
diff --git a/src/main/java/org/bukkit/craftbukkit/Main.java b/src/main/java/org/bukkit/craftbukkit/Main.java
index d665dc3fbd0989eb399c55d0717bc6fd79a0db17..91f031e3810744d66d0c5ea885ab33fd67d3d22d 100644
--- a/src/main/java/org/bukkit/craftbukkit/Main.java
+++ b/src/main/java/org/bukkit/craftbukkit/Main.java
@@ -177,6 +177,14 @@ public class Main {
                     .describedAs("Yml file");
                 // Purpur end - Purpur config files
 
+                // Osmium start
+                this.acceptsAll(asList("osmium", "osmium-settings"), "File for osmium settings")
+                    .withRequiredArg()
+                    .ofType(File.class)
+                    .defaultsTo(new File("osmium.yml"))
+                    .describedAs("Yml file");
+                // Osmium end
+
                 this.accepts("server-name", "Name of the server")
                         .withRequiredArg()
                         .ofType(String.class)
