From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Djatlasv <djatlasv@users.noreply.github.com>
Date: Fri, 13 Mar 2026 17:49:18 -0600
Subject: [PATCH] Osmium: rebrand BRAND_PURPUR_ID to BRAND_OSMIUM_ID


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfo.java b/src/main/java/io/papermc/paper/ServerBuildInfo.java
index fb1fe2651e53a9bf46b3632c638e13eea9dcda93..775d77c1d05c92bead2c183193994367f48eccb2 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfo.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfo.java
@@ -19,12 +19,12 @@ public interface ServerBuildInfo {
      */
     Key BRAND_PAPER_ID = Key.key("papermc", "paper");
 
-    // Purpur start
+    // Osmium start
     /**
-     * The brand id for Purpur.
+     * The brand id for Osmium.
      */
-    Key BRAND_PURPUR_ID = Key.key("purpurmc", "purpur");
-    // Purpur end
+    Key BRAND_OSMIUM_ID = Key.key("osmium", "osmium");
+    // Osmium end
     /**
      * Gets the {@code ServerBuildInfo}.
      *
