From 56a0a83ddb1b7472dbc9762e2aca396605c62085 Mon Sep 17 00:00:00 2001
From: File <noreply+automated@papermc.io>
Date: Sun, 20 Apr 1997 05:37:42 -0800
Subject: [PATCH] purpur File Patches


diff --git a/net/minecraft/world/level/block/CarvedPumpkinBlock.java b/net/minecraft/world/level/block/CarvedPumpkinBlock.java
index b45e5f571e72028996f641157473039ba9dd3520..b192b384fa5b9b6ab5b72f6c6414c4fa4f36c75e 100644
--- a/net/minecraft/world/level/block/CarvedPumpkinBlock.java
+++ b/net/minecraft/world/level/block/CarvedPumpkinBlock.java
@@ -67,7 +67,7 @@ public class CarvedPumpkinBlock extends HorizontalDirectionalBlock {
         if (blockPatternMatch != null) {
             SnowGolem snowGolem = EntityType.SNOW_GOLEM.create(level, EntitySpawnReason.TRIGGERED);
             if (snowGolem != null) {
-                spawnGolemInWorld(level, blockPatternMatch, snowGolem, blockPatternMatch.getBlock(0, 2, 0).getPos());
+                spawnGolemInWorld(level, blockPatternMatch, snowGolem, blockPatternMatch.getBlock(0, 2, 0).getPos(), this.placer); // Purpur - Summoner API
                 return;
             }
         }
@@ -77,7 +77,7 @@ public class CarvedPumpkinBlock extends HorizontalDirectionalBlock {
             IronGolem ironGolem = EntityType.IRON_GOLEM.create(level, EntitySpawnReason.TRIGGERED);
             if (ironGolem != null) {
                 ironGolem.setPlayerCreated(true);
-                spawnGolemInWorld(level, blockPatternMatch1, ironGolem, blockPatternMatch1.getBlock(1, 2, 0).getPos());
+                spawnGolemInWorld(level, blockPatternMatch1, ironGolem, blockPatternMatch1.getBlock(1, 2, 0).getPos(), this.placer); // Purpur - Summoner API
                 return;
             }
         }
@@ -86,7 +86,7 @@ public class CarvedPumpkinBlock extends HorizontalDirectionalBlock {
         if (blockPatternMatch2 != null) {
             CopperGolem copperGolem = EntityType.COPPER_GOLEM.create(level, EntitySpawnReason.TRIGGERED);
             if (copperGolem != null) {
-                spawnGolemInWorld(level, blockPatternMatch2, copperGolem, blockPatternMatch2.getBlock(0, 0, 0).getPos());
+                spawnGolemInWorld(level, blockPatternMatch2, copperGolem, blockPatternMatch2.getBlock(0, 0, 0).getPos(), this.placer); // Purpur - Summoner API
                 if (!copperGolem.valid) return; // Paper - entityspawnevent - entity was not added to the world so prevent world mutation
                 this.replaceCopperBlockWithChest(level, blockPatternMatch2);
                 copperGolem.spawn(this.getWeatherStateFromPattern(blockPatternMatch2));
@@ -105,7 +105,20 @@ public class CarvedPumpkinBlock extends HorizontalDirectionalBlock {
                 .getAge();
     }
 
+    @io.papermc.paper.annotation.DoNotUse
     private static void spawnGolemInWorld(Level level, BlockPattern.BlockPatternMatch patternMatch, Entity golem, BlockPos pos) {
+        // Purpur start - Summoner API
+        spawnGolemInWorld(level, patternMatch, golem, pos, null);
+    }
+    private static void spawnGolemInWorld(Level level, BlockPattern.BlockPatternMatch patternMatch, Entity golem, BlockPos pos, net.minecraft.world.entity.LivingEntity placer) {
+        java.util.UUID summoner = placer == null ? null : placer.getUUID();
+        switch (golem) {
+            case SnowGolem snowGolem -> snowGolem.setSummoner(summoner);
+            case IronGolem ironGolem -> ironGolem.setSummoner(summoner);
+            case CopperGolem copperGolem -> copperGolem.setSummoner(summoner);
+            default -> throw new IllegalStateException("Unexpected value: " + golem);
+        }
+        // Purpur end - Summoner API
         // clearPatternBlocks(level, patternMatch); // Paper - moved down
         golem.snapTo(pos.getX() + 0.5, pos.getY() + 0.05, pos.getZ() + 0.5, 0.0F, 0.0F);
         // Paper start
